### PR TITLE
[#18] Unit Tests for CountdownTimer Feature

### PR DESCRIPTION
**Pull request by anthonyle17**
Includes several unit tests for countdown timer feature #16 

Authors:
@anthonyle17 
@laurawheatley 

The last unit test at the bottom of the class is complete but should be implemented otherwise-- comment is provided for desired way.

### CODE REVIEW

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+        println("Testing if mCountdownTime is initialized as 0")
+        mMainActivity?.unsetCountdownMode()
+        //  Asserting that mCountdownTime has been set to 0
+        Assert.assertEquals(0, mMainActivity?.mCountdownTime)
+    }
+
+    @Test
+    fun unsetCountdownMode_mIsInCountdownModeTest() {
+        println("Testing if mIsInCountdownMode is false")
+    }
```
Comment (aa-software2112) [inline app/src/test/kotlin/CountdownTimerJUnitTest.kt:130]: This test is empty, please delete it

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        Assert.assertNotNull(countdownDropdownLinearLayout)
+
+        /**  Debugging purpose: in View.java, VISIBLE = 0x00000000 and INVISIBLE = 0x00000004
+
+        if (countdownDropdownLinearLayout?.visibility == 4) {
+        println("invisible")
+        }
+
+            countdownDropdownLinearLayout IS INVISIBLE
+         */
```
Comment (aa-software2112) [inline app/src/test/kotlin/CountdownTimerJUnitTest.kt]: remove this comment block

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
+        /**  Debugging purpose: in View.java, VISIBLE = 0x00000000 and INVISIBLE = 0x00000004
+
+        if (countdownDropdownLinearLayout?.visibility == 0) {
+        println("visible")
+        }
+
+            countdownDropdownLinearLayout IS INDEED VISIBLE
+         */
+    }
```
Comment (aa-software2112) [inline app/src/test/kotlin/CountdownTimerJUnitTest.kt:172]: remove this comment block

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt (hunk 1 above)
Comment (anthonyle17) [reply app/src/test/kotlin/CountdownTimerJUnitTest.kt:130]: Will implement it immediately

**Review by aa-software2112 — CHANGES_REQUESTED** [review]
See comments for changes, very minor fixes

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt (hunk 2 above)
Comment (anthonyle17) [reply app/src/test/kotlin/CountdownTimerJUnitTest.kt]: No problem!

File: app/src/test/kotlin/CountdownTimerJUnitTest.kt (hunk 3 above)
Comment (anthonyle17) [reply app/src/test/kotlin/CountdownTimerJUnitTest.kt:172]: No problem!

**Comment by aa-software2112** [comment]
The coverage with these tests is nice, good work @anthonyle17 @laurawheatley. Certainly if some of the implementation changes, these tests will catch it and let us know. Nice job!

**Comment by aa-software2112** [comment]
If Travis passes, go ahead and merge @anthonyle17 @laurawheatley

**Comment by anthonyle17** [comment]
@aa-software2112 Waiting impatiently hehe. Thanks!

**Review by aa-software2112 — APPROVED** [review]

**Comment by lkadian** [comment]
Nicely done -- went over the tests and they cover the implementation of the countdown feature well

**Review by lkadian — APPROVED** [review]

**Pull request merged**